### PR TITLE
Platform api

### PR DESCRIPTION
**Pull request by saibotma**

### CODE REVIEW

File: src/main/kotlin/platformapi/channelDetails.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    when (database.transaction {
+        val insertResult = insertChannel(channel.meta)
+        insertMembers(channel.members)
+        insertResult
+    }) {
+        is Fallible.Failure -> {
+            database.transaction {
+                updateChannel(channel.meta)
+                deleteMembersOf(location.channelId)
+                insertMembers(channel.members)
```
Comment (JulianBissekkou) [inline src/main/kotlin/platformapi/channelDetails.kt:29]: I am just scrolling through this out of curiosity but shouldn't you check for `InsertChannelError.Duplicate` in that case? 🤔

File: src/main/kotlin/platformapi/channelDetails.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    // Have two transactions as postgres does not like reusing a transaction that
+    // had errors. (https://stackoverflow.com/questions/10399727/psqlexception-current-transaction-is-aborted-commands-ignored-until-end-of-tra)
+    when (database.transaction {
+        val insertResult = insertChannel(channel.meta)
+        insertMembers(channel.members)
+        insertResult
+    }) {
+        is Fallible.Failure -> {
+            database.transaction {
+                updateChannel(channel.meta)
```
Comment (JulianBissekkou) [inline src/main/kotlin/platformapi/channelDetails.kt:27]: What if this fails? 🤔  You would still return `NoContent`

Comment (JulianBissekkou) [reply src/main/kotlin/platformapi/channelDetails.kt:27]: never mind, just saw there was a todo about finalizing the api of this method 🙈

File: src/main/kotlin/platformapi/models/ChannelMeta.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package platformapi.models
+
+import java.util.*
+
+data class ChannelMeta(
+    val id: UUID,
+    val name: String?,
+    val isManaged: Boolean,
```
Comment (JulianBissekkou) [inline src/main/kotlin/platformapi/models/ChannelMeta.kt:8]: What's a managed channel? :D

File: src/main/resources/application.conf [hunk 4]
```diff
@@ -20,3 +20,13 @@ postgres {
...
   db = ${?POSTGRES_DB}
 }
+
+platformApi {
+  accessToken = "a very bad access token"
+  accessToken = ${?PLATFORM_API_ACCESS_TOKEN}
+}
+
+clientApi {
+  jwtSecret = "a very bad secret"
```
Comment (JulianBissekkou) [inline src/main/resources/application.conf:30]: 😉

File: src/test/kotlin/platformapi/ChannelDetailsTests.kt [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                    shouldHaveSize(1)
+                    first() shouldBe updatedChannel
+                }
+            }
+        }
+    }
+}
+
+fun DI.MainBuilder.setupTestDependencies() {
+    val postgresContainer = PostgreSQLContainer<Nothing>()
```
Comment (JulianBissekkou) [inline src/test/kotlin/platformapi/ChannelDetailsTests.kt:66]: What is the generic type here? :O (normally)

File: src/test/kotlin/platformapi/ChannelDetailsTests.kt [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    withTestApplication({
+        module()
+        subDI(parentDI = closestDI()) {
+            setupTestDependencies()
+            bindDependencies()
+        }
+    }) {
+        val kodein = application.closestDI()
+        val objectMapper: ObjectMapper by kodein.instance()
+        print(objectMapper.writeValueAsString("klajsldkjfasdf"))
```
Comment (JulianBissekkou) [inline src/test/kotlin/platformapi/ChannelDetailsTests.kt:91]: debug artifact?

File: src/main/kotlin/platformapi/channelDetails.kt (hunk 1 above)
Comment (saibotma) [reply src/main/kotlin/platformapi/channelDetails.kt]: Nice catch, didn't notice that.

File: src/main/kotlin/platformapi/models/ChannelMeta.kt (hunk 3 above)
Comment (saibotma) [reply src/main/kotlin/platformapi/models/ChannelMeta.kt:8]: A managed channel is a channel that can only be altered by the platform API, but not by users.

File: src/test/kotlin/platformapi/ChannelDetailsTests.kt (hunk 5 above)
Comment (saibotma) [reply src/test/kotlin/platformapi/ChannelDetailsTests.kt:66]: It's a recursive generic type. 
`public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends JdbcDatabaseContainer<SELF>`. Do you have any idea on how to make this work?

File: src/test/kotlin/platformapi/ChannelDetailsTests.kt (hunk 6 above)
Comment (saibotma) [reply src/test/kotlin/platformapi/ChannelDetailsTests.kt]: Yes, thanks.

**Pull request merged**